lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,44 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
       @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      stoppable_call(20) { sleep(0.25) }
```

Comment:
actually, why is it necessary to go to such lengths here? it will be quite rare to hit stop! during sleep, so we can just double check:

```ruby
def run
  start_listener
rescue => e
  sleep(5) unless !stop
  retry unless !stop
end
```